frontend/apps/pablo/defi/hooks/pools/usePoolRatio.ts

```diff
@@ -1,23 +1,43 @@
-import { useEffect, useMemo, useState } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
 import type { GetStatsReturn } from "../../utils/pablo/pools/stats";
 import { getPoolVolume, getStats } from "../../utils/pablo/pools/stats";
 import useStore from "@/store/useStore";
-import { PoolConfig } from "@/store/pools/types";
 import BigNumber from "bignumber.js";
 import { getOraclePrice } from "@/store/oracle/slice";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
 
-export const usePoolRatio = (pool: PoolConfig) => {
+export const usePoolRatio = (poolId: string) => {
   const userOwnedLiquidity = useStore((store) => store.ownedLiquidity.tokens);
-  const balance = userOwnedLiquidity[pool.config.lpToken]?.balance ?? {
-    free: new BigNumber(0),
-    locked: new BigNumber(0),
-  };
```

Comment:
Is there any reason why passing `poolId` from the pool on the calling side and find pool again here?

Reply:
I wanted to have finer control over the pool, especially since it is an object, to track whether it is actually changed. A primitive poolId was easier to automatically take care of this.

Reply:
Although, your comment inspires me to tackle this at an upper level. Let me see how it goes.

Reply:
Alright, I think addressing this at the upper level requires more time and effort to properly tackle, within the current priorities, I'm not sure it is a good idea.

Reply:
Ok. Let's keep it as is. 🙂